File: java/1.4 Iteradores y Colecciones de Objetos/1.4.2 Iteradores/Ejemplo06/mx/uaemex/controller/ProductoController.java
package mx.uaemex.controller;

import mx.uaemex.data.Producto;
import java.util.Vector;
import java.util.Iterator;

public class ProductoController
{   
    private Vector<Producto> v = new Vector<Producto>();
    
    public ProductoController() {
        v.add(new Producto(1,"Tenis","Calzado deportivo en diversas tallas",2000.0f));
        v.add(new Producto(2,"Cartera","Cartera en piel para dama",1000.0f));
        v.add(new Producto(3,"Pants","Ropa deportiva unisex",950.0f));
        v.add(new Producto(4,"Gorra","Gorra de teporada",200.0f));
        verCatalogo();
        borrarProducto(2);
        verCatalogo();
    }

    public void borrarProducto(int id) {
        Iterator<Producto> it = v.iterator();
        while ( it.hasNext() ) {
            if ( it.next().getId() == id ) {
                it.remove();
            }
        }
    }
    
    
    public void verCatalogo() {
        //Uso de un Iterador para recorrer toda una colección de forma muy eficiente!
        Iterator<Producto> it = v.iterator();
        while ( it.hasNext() ) {
            it.next().getInfo();
        }
    }
    
    public static void main(String args[]) {
        new ProductoController();
    }
}

File: c++/1.4 Iteradores y Colecciones de Objetos/1.4.1 Arreglos/Ejemplo02/ProductoController.cpp
#include <iostream>
#include "Producto.h"

using namespace std;

class ProductoContoller {
  private:
    Producto *p[4];

  public:
    ProductoContoller() {
      p[0] = new Producto(1, "Tenis", "Calzado deportivo en diversas tallas", 2000.0f);
      p[1] = new Producto(2, "Cartera", "Cartera en piel para dama", 1000.0f);
      p[2] = new Producto(3, "Pants", "Ropa deportiva unisex", 950.0f);
      p[3] = new Producto(4, "Gorra", "Gorra de teporada", 200.0f);
      verCatalogo();
    }

    void verCatalogo() {
      int length = sizeof(p) / sizeof(p[0]);
      for (int i = 0; i < length; i++)
      {
        p[i]->getInfo();
      }
    }
};

int main(int argc, char const *argv[]) {
  ProductoContoller productoController;
  return 0;
}
